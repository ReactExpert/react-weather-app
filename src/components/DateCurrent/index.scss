.date-current {
  align-items: center;
  color: rgba(255, 255, 255, .8);
  display: flex;
  font-size: 18px;
  font-size: 16px;
  justify-content: center;
  left: 50%;
  position: absolute;
  text-align: center;
  transform: translate(-50%, 80px);
  width: 100%;

  &__text {
    opacity: 0;
    transform: translateY(-50px);
    transition: transform .3s cubic-bezier(.165, .84, .44, 1), opacity .3s .1s ease-out;
    will-change: opacity, transform;
  }

  &__text.animate-in {
    opacity: 1;
    transform: translateY(0);
  }
}
