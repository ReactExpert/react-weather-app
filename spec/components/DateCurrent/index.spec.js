import React from 'react';
import DateCurrent from '../../../src/components/DateCurrent';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

describe('DateCurrent', () => {
  it('matches snapshot', () => {
    const component = mount(<DateCurrent date='Sunday, July 1, 2018'/>);

    expect(component).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls animate', (done) => {
      const component = mount(<DateCurrent date='Sunday, July 1, 2018'/>);
      const instance = component.instance();

      instance.animate = jest.fn();

      setTimeout(() => {
        expect(instance.animate).toBeCalled();
        done();
      }, 400);
    });
  })
});
