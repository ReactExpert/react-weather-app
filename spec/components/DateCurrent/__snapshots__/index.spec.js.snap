// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateCurrent matches snapshot 1`] = `
<DateCurrent
  date="Sunday, July 1, 2018"
>
  <section
    className="date-current"
  >
    <span
      className="date-current__text"
    >
      Sunday, July 1, 2018
    </span>
  </section>
</DateCurrent>
`;
